skbio/io/format/genbank.py

```diff
@@ -0,0 +1,866 @@
+"""
+GenBank format (:mod:`skbio.io.format.genbank`)
+===============================================
+
+.. currentmodule:: skbio.io.format.genbank
+
+GenBank format (GenBank Flat File Format) stores sequence and its annotation
+together. The start of the annotation section is marked by a line beginning
+with the word "LOCUS". The start of sequence section is marked by a line
+beginning with the word "ORIGIN" and the end of the section is marked by a line
+with only "//".
+
+The GenBank file usually ends with .gb or sometimes .gbk. The GenBank format
+for protein has been renamed to GenPept. The GenBank (for nucleotide) and
+Genpept are essentially the same format.
+
+An example of a GenBank file can be see here:
+<http://www.ncbi.nlm.nih.gov/Sitemap/samplerecord.html>
+
+Format Support
+--------------
+**Has Sniffer: Yes**
+
++------+------+---------------------------------------------------------------+
+|Reader|Writer|                          Object Class                         |
++======+======+===============================================================+
+|Yes   |Yes   |:mod:`skbio.sequence.Sequence`                                 |
++------+------+---------------------------------------------------------------+
+|Yes   |Yes   |:mod:`skbio.sequence.DNA`                                      |
++------+------+---------------------------------------------------------------+
+|Yes   |Yes   |:mod:`skbio.sequence.RNA`                                      |
++------+------+---------------------------------------------------------------+
+|Yes   |Yes   |:mod:`skbio.sequence.Protein`                                  |
++------+------+---------------------------------------------------------------+
+|Yes   | Yes  | generator of :mod:`skbio.sequence.Sequence` objects           |
++------+------+---------------------------------------------------------------+
+
+Format Specification
+--------------------
+**State: Experimental as of 0.4.0-dev.**
+
+The International Nucleotide Sequence Database Collaboration (INSDC)
+foundational initiative between the DDBJ, EMBL, and GenBank
+(http://www.insdc.org/). These organisations all use the
+same "Feature Table" layout in their plain text flat file formats.
+
+However, the header and sequence sections of an EMBL file are very
+different in layout to those produced by GenBank/DDBJ.
+
+Feature Table Documentation:
+http://www.insdc.org/files/feature_table.html
+ftp://ftp.ncbi.nih.gov/genbank/docs/FTv10_3.html
+
+The sequence in the ``'ORIGIN'`` section is always in lowercase for the
+GenBank files downloaded from NCBI. For the RNA molecules, ``'t'`` (thymine),
+instead of ``'u'`` (uracil) is used in the sequence. All GenBank writers
+follow these conventions while writing GenBank files.
+
```

Comment:
Can you describe here what the structure of the metadata will look like and any relationships it may have with positional metadata.